hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -343,15 +343,15 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
       sendRejectionReasonToRingBuffer(() -> getBalanceReason(calculatedTotal, calculatedMultiplier),
         costFunctions);
     }
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("{} {}; total cost={}, sum multiplier={}; cost/multiplier to need a balance is {}",
-          balanced ? "Skipping load balancing because balanced" : "We need to load balance",
-          isByTable ? String.format("table (%s)", tableName) : "cluster",
-          total, sumMultiplier, minCostNeedBalance);
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("Balance decision detailed function costs={}", functionCost());
-      }
+    LOG.info("{} {}; total cost={}, sum multiplier={}; cost/multiplier to need a balance is {}",
+      balanced ? "Skipping load balancing because balanced" :
```

Comment:
If balanced, shouldn't it say why it thinks its balanced?

Reply:
you mean beyond the given cost, multiplier, and needed threshold? you thinking of the individual cost functions?

Reply:
I'm thinking of the times when balancer says the cluster is balanced but when when the operator looks at the Master UI, it plainly is not.... For example, if the regions-per-server are not aligned, this message would explain why it thinks the cluster balanced though to the the operator, the cluster looks off.

Reply:
There is a function that explain the reason of skipping. But we need to provide the big picture why t stops here. Will update.

Reply:
https://issues.apache.org/jira/browse/HBASE-25666

Reply:
For the Stochastic Load Balancer isn't the only way to represent that dumping the cost functions? We do that at the more granular log level. You thinking we need to do it all the time @saintstack ?

Reply:
I was thinking either explaining terms or pointing to doc or elsewhere where terms are explained and how the balancer machinations engage w/ them. Can do in a later PR, yes. This one is a big improvement as is.